In [1]:
from utils.preprocessing import *
from utils.swSets import *
import matplotlib.pyplot as plt
import pickle

## <font color="rebeccapurple"> Corpus

In [5]:
#path = "/Vrac/PLDAC_reco/shows"

In [6]:
#corpus = corpus_shows(path)

In [9]:
#pickle.dump(corpus, open("/Vrac/PLDAC_reco/pickles/corpus.p", "wb"))

In [2]:
corpus = pickle.load(open("/Vrac/PLDAC_reco/pickles/corpus.p", "rb"))

## <font color="rebeccapurple"> Taille du vocabulaire

In [7]:
mat = getTfidfSparseMat(corpus, my_stopwords=stopwords_set)

taille vocabulaire :  561922


La taille du vocabulaire sans autre traitement que les stopwords est de 561922

In [11]:
pickle.dump(mat, open("/Vrac/PLDAC_reco/pickles/mat.p", "wb"))

## <font color="rebeccapurple"> test differents paramètres

### <font color="darkviolet"> max_features=50000, min_df=20, max_df=0.4

In [12]:
mat = getTfidfSparseMat(corpus, my_stopwords=stopwords_set,max_features=50000, min_df=20, max_df=0.4)

taille vocabulaire :  50000


In [13]:
pickle.dump(mat, open("/Vrac/PLDAC_reco/pickles/mat_voc_50000_mindf_20_max_df_0.4.p", "wb"))

### <font color="darkviolet">min_df=20, max_df=0.4

In [16]:
mat = getTfidfSparseMat(corpus, my_stopwords=stopwords_set, min_df=20, max_df=0.4)

taille vocabulaire :  53540


In [17]:
pickle.dump(mat, open("/Vrac/PLDAC_reco/pickles/mat_voc_53540_mindf_20_max_df_0.4.p", "wb"))

**remarque**: utiliser un min_df de 20 et un max df de 0.4 permettent déjà de réduire le vocabulaire de façon conséquente :  561922 --->  53540

### <font color="darkviolet"> max_features=25000, min_df=20, max_df=0.4

In [18]:
mat = getTfidfSparseMat(corpus, my_stopwords=stopwords_set, max_features=25000, min_df=20, max_df=0.4)

taille vocabulaire :  25000


In [19]:
pickle.dump(mat, open("/Vrac/PLDAC_reco/pickles/mat_voc_25000_mindf_20_max_df_0.4.p", "wb"))

### <font color="darkviolet">min_df=3, max_df=0.4

In [3]:
mat = getTfidfSparseMat(corpus, my_stopwords=stopwords_set, min_df=3, max_df=0.4)

taille vocabulaire :  141418


In [4]:
pickle.dump(mat, open("/Vrac/PLDAC_reco/pickles/mat_voc_141418_mindf_3_max_df_0.4.p", "wb"))

### <font color="darkviolet">max_features = 50000, min_df=3, max_df=0.4

In [5]:
mat = getTfidfSparseMat(corpus, my_stopwords=stopwords_set, max_features=50000, min_df=3, max_df=0.4)

taille vocabulaire :  50000


In [6]:
pickle.dump(mat, open("/Vrac/PLDAC_reco/pickles/mat_voc_50000_mindf_3_max_df_0.4.p", "wb"))

### <font color="darkviolet">max_features = 50000, min_df=10, max_df=0.4

In [7]:
mat = getTfidfSparseMat(corpus, my_stopwords=stopwords_set, max_features=50000, min_df=10, max_df=0.4)

taille vocabulaire :  50000


In [8]:
pickle.dump(mat, open("/Vrac/PLDAC_reco/pickles/mat_voc_50000_mindf_10_max_df_0.4.p", "wb"))

### <font color="darkviolet">max_features = 50000, min_df=15, max_df=0.4

In [9]:
mat = getTfidfSparseMat(corpus, my_stopwords=stopwords_set, max_features=50000, min_df=15, max_df=0.4)

taille vocabulaire :  50000


In [10]:
pickle.dump(mat, open("/Vrac/PLDAC_reco/pickles/mat_voc_50000_mindf_15_max_df_0.4.p", "wb"))

### <font color="darkviolet">max_features = 50000, min_df=5, max_df=0.4

In [11]:
mat = getTfidfSparseMat(corpus, my_stopwords=stopwords_set, max_features=50000, min_df=5, max_df=0.4)

taille vocabulaire :  50000


In [12]:
pickle.dump(mat, open("/Vrac/PLDAC_reco/pickles/mat_voc_50000_mindf_5_max_df_0.4.p", "wb"))